model/storage/disk/disk_model_store.py

```diff
@@ -34,17 +36,32 @@ def store_model(self, hotkey: str, model: Model) -> ModelId:
         # Return the same model id used as we do not edit the commit information.
         return model.id
 
-    def retrieve_model(self, hotkey: str, model_id: ModelId) -> Model:
-        """Retrieves a trained model locally."""
-
-        model = AutoModelForCausalLM.from_pretrained(
-            pretrained_model_name_or_path=utils.get_local_model_snapshot_dir(
-                self.base_dir, hotkey, model_id
-            ),
-            revision=model_id.commit,
-            local_files_only=True,
-            use_safetensors=True,
-        )
+    def retrieve_model(
+        self, hotkey: str, model_id: ModelId, optimized: bool = False
```

Comment:
Fine with this as-is, but I'm wondering if we need to gate this behind a block or whether we should just always do this? It looks like the current list of allowed models is a subset of the models allowed at 7B, so bfloat16 and flash attention should work.

One other note that I just remembered, I _think_ flash attention only works on some subset of GPUs. We should probably note this somewhere in docs and probably announce it to valis: https://github.com/Dao-AILab/flash-attention

Reply:
I removed GPT2 and GPTJ. This is also why I had to update the default model for miners.

Reply:
Added to docs about flash attention